### PR TITLE
raftstore: add ReadExecutor to execute Get/Snap commands

### PR DESCRIPTION
**Pull request by overvenus**
Together read functions to a struct, so It is easier to reuse in local reader.

Cc #2639

### CODE REVIEW

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -1930,27 +1931,75 @@ impl Peer {
...
                 CmdType::Snap => {
                     need_snapshot = true;
                     raft_cmdpb::Response::new()
                 }
                 CmdType::Prewrite
                 | CmdType::Put
                 | CmdType::Delete
-                | CmdType::DeleteRange
                 | CmdType::IngestSST
+                | CmdType::DeleteRange
```
Comment (BusyJay) [inline src/raftstore/store/peer.rs]: Why move it around?

File: src/raftstore/store/peer.rs [hunk 2]
```diff
@@ -1930,27 +1931,75 @@ impl Peer {
...
+    fn new(region: &'r metapb::Region, engine: &'e Arc<DB>, tag: &'t str) -> Self {
+        ReadExecutor {
+            region,
+            engine,
+            tag,
+        }
+    }
+
+    fn do_get(&self, req: &Request, snap: &Snapshot) -> Result<Response> {
+        // TODO: the get_get looks wried, maybe we should figure out a better name later.
```
Comment (BusyJay) [inline src/raftstore/store/peer.rs]: s/wried/weird/

**Comment by overvenus** [comment]
PTAL

File: src/raftstore/store/worker/apply.rs [hunk 3]
```diff
@@ -1599,7 +1599,8 @@ impl ApplyDelegate {
 
     fn handle_put(&mut self, ctx: &ApplyContext, req: &Request) -> Result<Response> {
         let (key, value) = (req.get_put().get_key(), req.get_put().get_value());
-        check_data_key(key, &self.region)?;
+        // region key range has no data prefix, so we must use origin key to check.
```
Comment (huachaohuang) [inline src/raftstore/store/worker/apply.rs]: I think it is not necessary to add this comment everywhere, maybe just add it to the function documentation comment.

Comment (overvenus) [reply src/raftstore/store/worker/apply.rs]: Well, I think it's better to be verbose. We can remove them if you insist.

**Pull request merged**